src/service/allocator/allocator.service.ts

```diff
@@ -19,11 +19,24 @@ export class AllocatorService {
   ) {}
 
   async getAllocatorRetrievability() {
+    const lastWeek =
+      await this.prismaService.allocators_weekly.findFirstOrThrow({
+        select: {
+          week: true,
+        },
+        orderBy: {
+          week: 'desc',
+        },
+      });
+
     const averageSuccessRate =
       await this.prismaService.allocators_weekly.aggregate({
         _avg: {
           avg_weighted_retrievability_success_rate: true,
         },
+        where: {
+          week: lastWeek.week,
```

Comment:
We should be taking only last full week. So if we're on Tuesday Oct 1, we should take week Sep 23 - Sep 29.
This query would take week Sep 30 -  Oct 6, which isn't full and isn't a good week to take average from.